server.js

```
@@ -0,0 +1,218 @@
+        if (files.includes(req.url.slice(1)) === false) { // Directory doesn't exist
+          if (req.url === "/") { // For default page
+            directory = "/index.html";
+            contentType = "html"
+          } else if (req.url.slice(-3) === "css") { // Needed to load css properly
+            directory = `${req.url}`
+            contentType = "css"
+          } else {
+            directory = '/404.html';
+            contentType = 'html';
+          }
+        } else { // Directory exists
+          if (req.url.slice(-3) === "css") { // Needed to load css properly
+            directory = `${req.url}`
+            contentType = "css"
+          } else {
+            directory = req.url
+            contentType = req.url.slice(req.url.indexOf('.') + 1, req.url.length);
+          }
+        }
+        fs.readFile(`./public${directory}`, (err, data) => { // Sends requested data
+          if (err) {
+            return console.log(err);
+          }
+          res.writeHead(200, {
+            "Content-Type": `text/${contentType}, charset=utf-8`,
+            "Content-Length": data.toString().length,
+            Date: date
+          });
+          res.write(data.toString());
+          res.end();
+        });
+      });
+    }
+    /*** POST Method ***/
+    if (req.method === "POST") {
+      // Decode from the key value pairs in Postman
+      let decodeChunk = querystring.parse(body);
+      // Used for new element pages
+      addOrUpdateElem(decodeChunk, req, res)
+      // Update index.html
+      parseNewHTML(decodeChunk, req, res);
+      res.writeHead(200, {
+        "Content-Type": 'application/json',
+        "Success": true,
+        "Date": date
+      })
+      res.end();
+    }
+    /*** PUT Method ***/
+    if (req.method === "PUT") {
+      // Decode from the key value pairs in Postman
+      let decodeChunk = querystring.parse(body);
+      fs.readdir("./public/", (err, files) => {
+        if (err) {
+          console.log(err)
```

This should return an error to the client using either a 400/500 error